tools/alive_lexer.re

```diff
@@ -104,6 +104,16 @@ re2c:yyfill:check = 0;
   return INT_TYPE;
 }
 
+[-+]? [0-9]* "."? [0-9]+ ([eE] [-+]? [0-9]+)? "d" {
+  yylval.double_num = strtod((char*)YYTEXT, nullptr);
+  return DOUBLE_NUM;
+}
+
+[-+]? [0-9]* "."? [0-9]+ ([eE] [-+]? [0-9]+)? "f" {
```

Comment:
it's unclear to me why we need this distinction. If a user wants a specific type, they can write the type f64/f32/f16/whatever, like we do with integers. Can we always parse the number as double?

Reply:
I am thinking about parsing constants some larger FP types like fp80 and fp128. Just like the suffix "l"(long), "ll"(long long) in integer constants, user may reach larger floating point range by this distinction.

Also another possible solution is that we'll always parse the constants with the biggest type "fp128" and cast them to the corresponding type later.

Reply:
Ye, I think it should be transparent to the user

Reply:
fixed this :-)